zipline/data/hdf5_daily_bars.py

```diff
@@ -0,0 +1,758 @@
+    """
+    Parameters
+    ---------
+    country_group : h5py.Group
+        The group for a single country in an HDF5 daily pricing file.
+    """
+    def __init__(self, country_group):
+        self._country_group = country_group
+
+        self._postprocessors = {
+            OPEN: partial(convert_price_with_scaling_factor,
+                          scaling_factor=self._read_scaling_factor(OPEN)),
+            HIGH: partial(convert_price_with_scaling_factor,
+                          scaling_factor=self._read_scaling_factor(HIGH)),
+            LOW: partial(convert_price_with_scaling_factor,
+                         scaling_factor=self._read_scaling_factor(LOW)),
+            CLOSE: partial(convert_price_with_scaling_factor,
+                           scaling_factor=self._read_scaling_factor(CLOSE)),
+            VOLUME: lambda a: a,
+        }
+
+    @classmethod
+    def from_file(cls, h5_file, country_code):
+        """
+        Parameters
+        ----------
+        h5_file : h5py.File
+            An HDF5 daily pricing file.
+        country_code : str
+            The ISO 3166 alpha-2 country code for the country to read.
+        """
+        return cls(h5_file[country_code])
+
+    @classmethod
+    def from_path(cls, path, country_code):
+        """
+        Parameters
+        ----------
+        path : str
+            The path to an HDF5 daily pricing file.
+        country_code : str
+            The ISO 3166 alpha-2 country code for the country to read.
+        """
+        return cls.from_file(h5py.File(path), country_code)
+
+    def _read_scaling_factor(self, field):
+        return self._country_group[DATA][field].attrs[SCALING_FACTOR]
+
+    def load_raw_arrays(self,
+                        columns,
+                        start_date,
+                        end_date,
+                        assets):
+        """
+        Parameters
+        ----------
+        columns : list of str
+           'open', 'high', 'low', 'close', or 'volume'
+        start_date: Timestamp
+           Beginning of the window range.
+        end_date: Timestamp
+           End of the window range.
+        assets : list of int
+           The asset identifiers in the window.
+
+        Returns
+        -------
+        list of np.ndarray
+            A list with an entry per field of ndarrays with shape
+            (minutes in range, sids) with a dtype of float64, containing the
+            values for the respective field over start and end dt range.
+        """
+        start = start_date.asm8
+        end = end_date.asm8
+
+        sid_selector = self.sids.searchsorted(assets)
+
+        date_slice = self._compute_date_range_slice(start, end)
+        nrows = date_slice.stop - date_slice.start
+        out = []
+        for column in columns:
+            dataset = self._country_group[DATA][column]
+            ncols = dataset.shape[0]
+            shape = (ncols, nrows)
+            buf = np.full(shape, 0, dtype=np.uint32)
```

Comment:
Minor optimization note: we should use `np.zeros` instead of `np.full` here. Most modern operating systems have fast paths for returning zeroed pages, so I'd expect to get a small win here.